core/src/main/java/org/elasticsearch/index/query/BoolQueryBuilder.java

```diff
@@ -262,6 +263,13 @@ public String getWriteableName() {
 
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
+        final int numClauses = mustNotClauses.size() + filterClauses.size() + shouldClauses.size() + mustClauses.size();
+        if (numClauses == 1 && must().size() == 1 && boost() == DEFAULT_BOOST) {
+            // we really only optimize this for testing since we use this to rewrite
+            // named queries TemplateQueryBuilder and WrapperQueryBuilder.
+            // it's equivalent but will anyways happen later down the road in the BQ#rewrite method
+            return mustClauses.get(0).toQuery(context);
+        }
```

Comment:
should we instead move this special case to TemplateQueryBuilderTests and WrapperQueryBuilderTests doAssertLuceneQuery so that they assert on the right thing?